### PR TITLE
add JsonFraming usage example to the docs #24495

### LINKED ISSUE #24495: JsonFraming is missing docs
> I googled for some Ideas how to extract the JSON object efficiently. I stumbled across this: 
> https://doc.akka.io/japi/akka/current/akka/stream/javadsl/JsonFraming.html 
> And I couldn't find a single code example how to use it. 

He's right, we don't have docs about it in streams.
We only document JSON stuff in akka http, and there in the streaming framing thingies, which as separate API

### PR DESCRIPTION
**Pull request by shkoder**
resolves #24495

### CODE REVIEW

**Comment by ktoso** [comment]
thanks a lot!

**Comment by ktoso** [comment]
OK TO TEST

**Comment by shkoder** [comment]
Flaky test: `akka.stream.scaladsl.UnfoldResourceAsyncSourceSpec.Unfold Resource Async Source must close resource when stream is abruptly terminated`

Looks like #23382 resurfacing

**Comment by raboof** [comment]
PLS BUILD

File: akka-docs/src/main/paradox/stream/stream-io.md [hunk 1]
```diff
@@ -123,6 +123,17 @@ see
 @java[[Javadoc](http://doc.akka.io/japi/akka/current/akka/stream/javadsl/Framing.html#simpleFramingProtocol-int-)]
 for more information.
 
+`JsonFraming` (
+@scala[[ScalaDoc](http://doc.akka.io/api/akka/current/akka/stream/scaladsl/JsonFraming$.html)]
+@java[[Javadoc](http://doc.akka.io/japi/akka/current/akka/stream/javadsl/JsonFraming.html#objectScanner-int-)]
```
Comment (ktoso) [inline akka-docs/src/main/paradox/stream/stream-io.md]: Change this to:

```
@scala[[JsonFraming](http://doc.akka.io/api/akka/current/akka/stream/scaladsl/JsonFraming$.html)]@java[[JsonFraming](http://doc.akka.io/japi/akka/current/akka/stream/javadsl/JsonFraming.html#objectScanner-int-)]
```

File: akka-docs/src/main/paradox/stream/stream-io.md [hunk 2]
```diff
@@ -133,7 +144,7 @@ Scala
 :  @@snip [TcpSpec.scala]($akka$akka-stream-tests/src/test/scala/akka/stream/io/TcpSpec.scala) { #setting-up-ssl-context }
 
 Java
-:  @@snip [TcpSpec.scala]($akka$akka-stream-tests/src/test/java/akka/stream/javadsl/TcpTest.java) { #setting-up-ssl-context }
+:  @@snip [TcpTest.java]($akka$akka-stream-tests/src/test/java/akka/stream/javadsl/TcpTest.java) { #setting-up-ssl-context }
```
Comment (ktoso) [inline akka-docs/src/main/paradox/stream/stream-io.md]: good catch thanks

**Review by ktoso — APPROVED** [review]
Very nice, thanks

**Comment by ktoso** [comment]
LGTM once https://github.com/akka/akka/pull/24498#discussion_r165938128 is addressed

File: akka-docs/src/main/paradox/stream/stream-io.md (hunk 1 above)
Comment (raboof) [reply akka-docs/src/main/paradox/stream/stream-io.md]: See also #24426 :)

**Review by ktoso — APPROVED** [review]
Fixed the link, thanks!

**Pull request merged**